minio/credentials/providers.py

```diff
@@ -80,7 +81,20 @@ def _urlopen(
     """Wrapper of urlopen() handles HTTP status code."""
     res = http_client.urlopen(method, url, body=body, headers=headers)
     if res.status not in [200, 204, 206]:
-        raise ValueError(f"{url} failed with HTTP status code {res.status}")
+        splits = urlsplit(url)
+        safe_url = url_replace(
+            url=splits,
+            query="&".join(
+                [re.sub(
+                    r"LDAPPassword=([^/]+)",
+                    "LDAPPassword=*REDACTED*",
+                    param,
+                ) for param in splits.query.split("&")],
+            ),
+        )
+        raise ValueError(
+            f"{safe_url} failed with HTTP status code {res.status}",
+        )
```

Comment:
Couldn't we just redact the password directly in the URL string?
```suggestion
        safe_url = re.sub(
            r"LDAPPassword=([^&]+)",
            "LDAPPassword=*REDACTED*",
            url,
        )
        raise ValueError(
            f"{safe_url} failed with HTTP status code {res.status}",
        )
```

Reply:
Also, `url_replace` will return a `urllib.parse.SplitResult`, not an actual url. So currently, instead of 
```
ValueError: 127.0.0.1:9001?Action=AssumeRoleWithLDAPIdentity&Version=2011-06-15&LDAPUsername=ASIAWUJLRDEGNXB49ZU5&LDAPPassword=*REDACTED* failed with HTTP status code 400
```

I get this 
```
ValueError: SplitResult(scheme='', netloc='', path='127.0.0.1:9001', query='Action=AssumeRoleWithLDAPIdentity&Version=2011-06-15&LDAPUsername=ASIAWUJLRDEGNXB49ZU5&LDAPPassword=*REDACTED*', fragment='') failed with HTTP status code 400
```